### PR TITLE
Lightbox to header, add tooltip to buttons and small styling adjustments (Meta #119)

### PR DESCRIPTION
**Pull request by marcwieland95**

### CODE REVIEW

File: src/Resources/views/tailwind_2/_header.html.twig [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
+        <div class="mt-4 flex sm:mt-0 sm:ml-4 space-x-3 items-center {{ block is defined ? ' mr-3' : '' }}">
             {% if table.exporters|length > 0 and table.option('definition') and table.option('definition').hasCapability(constant('araise\\CrudBundle\\Enums\\Page::EXPORT')) %}
                 <div
                     {{ stimulus_controller('araise/core-bundle/dropdown') }}
                     class="relative inline-block">
                     <button
                         {{ stimulus_action('araise/core-bundle/dropdown', 'toggle') | stimulus_action('araise/core-bundle/dropdown', 'close', 'keydown.esc') }}
-                        class="whatwedo_table:header__button whatwedo_table:header__button--download whatwedo_table-button h-full" tabindex="-1"
+                        {{ stimulus_controller('araise/crud-bundle/tooltip', { 'title': 'araise_table.download.tooltip'|trans } ) }}
+                        class="whatwedo_table:header__button whatwedo_table:header__button--download whatwedo_table-button" tabindex="-1"
```
Comment (renestalder) [inline src/Resources/views/tailwind_2/_header.html.twig:19]: What's with the funny `tabindex="-1"` here? Are we blocking keyboard and screen reader users from using the tooltip? Why?

Also, let's fix that button too by adding a `sr-only` text into it.

File: src/Resources/views/tailwind_2/_header.html.twig [hunk 2]
```diff
@@ -106,15 +108,14 @@
                                 value="{{ table.searchExtension.query }}"
                             >
                         </div>
-                        <button class="whatwedo_table-button--search" type="submit">
-                            {{ bootstrap_icon('search', { class: 'h-4 w-4 text-neutral-400', alt: 'Suchen' }) }}
+                        <button class="whatwedo_table-button--search" type="submit" title="Suche">
```
Comment (renestalder) [inline src/Resources/views/tailwind_2/_header.html.twig:111]: Let's fix that button altogether:

1. Add an `sr-only` Text to the button. Buttons either have an `aria-label` or a text, where the former is a last resort to use.
2.  Keep the title for mouse users too, but make it translatable.

File: src/Resources/translations/messages.en.yaml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 araise_table:
     download:
+        tooltop: Export
```
Comment (renestalder) [inline src/Resources/translations/messages.en.yaml:3]: There is a typo. It should be `tooltip` instead of `tooltop`.

**Review by renestalder — CHANGES_REQUESTED** [review]

File: src/Resources/translations/messages.en.yaml (hunk 3 above)
Comment (marcwieland95) [reply src/Resources/translations/messages.en.yaml:3]: Oh, that's not supposed to be here. The correct translation is in the other bundle

File: src/Resources/views/tailwind_2/_header.html.twig [hunk 4]
```diff
@@ -98,17 +104,82 @@
...
+                                class="whatwedo-crud-modal__body w-full"
+                                {{ stimulus_target('araise/core-bundle/modal-form', 'modalBody') }}
+                            ></div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    {% endif %}
+{% endmacro %}
```
Comment (renestalder) [inline src/Resources/views/tailwind_2/_header.html.twig:184]: Nit: That macro should at least go to the bottom of the file. But the best approach would be, putting it into a separate file and importing it.

The current approach works, but is the messy one.

Another Nit is that the macro is using snake_case and the variables are using camelCase. The official coding standards of Twig say to use snake_case (See paragrpah `Use lower cased and underscored variable names`).

https://twig.symfony.com/doc/3.x/coding_standards.html

but this Nit is really out of scope as the overall code does not apply any coding standards and checks and is currently already mixed between snake_case and camelCase.

File: src/Resources/views/tailwind_2/_header.html.twig [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
+        <div class="mt-4 flex sm:mt-0 sm:ml-4 space-x-3 items-center {{ block is defined ? ' mr-3' : '' }}">
             {% if table.exporters|length > 0 and table.option('definition') and table.option('definition').hasCapability(constant('araise\\CrudBundle\\Enums\\Page::EXPORT')) %}
                 <div
                     {{ stimulus_controller('araise/core-bundle/dropdown') }}
                     class="relative inline-block">
                     <button
                         {{ stimulus_action('araise/core-bundle/dropdown', 'toggle') | stimulus_action('araise/core-bundle/dropdown', 'close', 'keydown.esc') }}
-                        class="whatwedo_table:header__button whatwedo_table:header__button--download whatwedo_table-button h-full" tabindex="-1"
+                        {{ stimulus_controller('araise/crud-bundle/tooltip', { 'title': 'araise_table.download.tooltip'|trans } ) }}
+                        class="whatwedo_table:header__button whatwedo_table:header__button--download whatwedo_table-button"
```
Comment (renestalder) [inline src/Resources/views/tailwind_2/_header.html.twig:19]: Nit: Still a button without a text or aria-label.

**Review by renestalder — APPROVED** [review]
@marcwieland95 lgtm for now.

**Comment by marcwieland95** [comment]
Blocked by https://github.com/araise-dev/CrudBundle/pull/75

**Pull request merged**